cache/blobs/compression.go

```diff
@@ -0,0 +1,122 @@
+package blobs
+
+import (
+	"bytes"
+	"context"
+	"io"
+
+	"github.com/containerd/containerd/content"
+	"github.com/containerd/containerd/images"
+	"github.com/moby/buildkit/cache"
+	digest "github.com/opencontainers/go-digest"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/pkg/errors"
+)
+
+// CompressionType represents compression type for blob data.
+type CompressionType int
+
+const (
+	// Uncompressed indicates no compression.
+	Uncompressed CompressionType = iota
+
+	// Gzip is used for blob data.
+	Gzip
+
+	// UnknownCompression means not supported yet.
+	UnknownCompression
```

Comment:
Can you reorder these so unknown is the first. If we add more in the future, wouldn't want `unknown` to mean `2`. Or probably even better if you set it to `-1`

Reply:
make senses. Updated